src/circuit/v2/stream-handler.ts

```diff
@@ -0,0 +1,82 @@
+import { logger } from '@libp2p/logger'
+import * as lp from 'it-length-prefixed'
+import { Handshake, handshake } from 'it-handshake'
+import type { Stream } from '@libp2p/interface-connection'
+import type { Source } from 'it-stream-types'
+
+const log = logger('libp2p:circuit:v2:stream-handler')
+
+export interface StreamHandlerOptions {
+  /**
+   * A duplex iterable
+   */
+  stream: Stream
+
+  /**
+   * max bytes length of message
+   */
+  maxLength?: number
+}
+
+export class StreamHandlerV2 {
```

Comment:
I think you can remove this class by using [it-pb-stream](https://www.npmjs.com/package/it-pb-stream) to read/write pb encoded messages from the stream.